src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java

```diff
@@ -765,170 +773,121 @@ public int getParameterCount() throws SQLServerException {
-                int n = (int) getParameterInfo(param).get("COLUMN_TYPE");
-                switch (n) {
-                    case 1:
-                        return parameterModeIn;
-                    case 2:
-                        return parameterModeOut;
-                    default:
-                        return parameterModeUnknown;
-                }
-            }
-        } catch (SQLException e) {
-            SQLServerException.makeFromDriverError(con, stmtParent, e.toString(), null, false);
-            return parameterModeUnknown;
+            else if (n == 2)
+                return parameterModeOut;
+            else
+                return parameterModeUnknown;
         }
     }
 
     @Override
     public int getParameterType(int param) throws SQLServerException {
         checkClosed();
+        checkParam(param);
         int parameterType = 0;
-        try {
-            if (procMetadata == null) {
-                // PreparedStatement.
-                checkParam(param);
-                parameterType = queryMetaMap.get(param).parameterType;
-            } else {
-                Map<String, Object> info = getParameterInfo(param);
-                if (null != info) {
-                    parameterType = (short) info.get("DATA_TYPE");
-                }
-            }
-            if (0 != parameterType) {
-                switch (parameterType) {
-                    case microsoft.sql.Types.DATETIME:
-                    case microsoft.sql.Types.SMALLDATETIME:
-                        parameterType = SSType.DATETIME2.getJDBCType().asJavaSqlType();
-                        break;
-                    case microsoft.sql.Types.MONEY:
-                    case microsoft.sql.Types.SMALLMONEY:
-                        parameterType = SSType.DECIMAL.getJDBCType().asJavaSqlType();
-                        break;
-                    case microsoft.sql.Types.GUID:
-                        parameterType = SSType.CHAR.getJDBCType().asJavaSqlType();
-                        break;
-                    default:
-                        break;
-                }
+        if (null == procMetadata) {
+            parameterType = queryMetaMap.get(param).parameterType;
+        } else {
+            parameterType = (short) getParameterInfo(param).get("DATA_TYPE");
+        }
+        if (0 != parameterType) {
+            switch (parameterType) {
+                case microsoft.sql.Types.DATETIME:
+                case microsoft.sql.Types.SMALLDATETIME:
+                    parameterType = SSType.DATETIME2.getJDBCType().asJavaSqlType();
+                    break;
+                case microsoft.sql.Types.MONEY:
+                case microsoft.sql.Types.SMALLMONEY:
+                    parameterType = SSType.DECIMAL.getJDBCType().asJavaSqlType();
+                    break;
+                case microsoft.sql.Types.GUID:
+                    parameterType = SSType.CHAR.getJDBCType().asJavaSqlType();
+                    break;
```

Comment:
Not a big deal but I would've preferred to see an explicit default case f does nothing but break just for completeness and maintainability.